crates/fluvio-controlplane-metadata/src/smartmodule/spec.rs

```diff
@@ -111,6 +118,35 @@ pub struct SmartModuleWasm {
     #[cfg_attr(feature = "use_serde", serde(with = "base64"))]
     pub payload: Vec<u8>,
 }
+
+impl SmartModuleWasm {
+    /// Validates a SmartModule's WASM payload to represent a valid WASM file
+    /// in the binary format (*.wasm).
+    pub fn validate_binary(mut data: &[u8]) -> Result<(), ValidateSmartModuleWasmError> {
+        let mut parser = Parser::default();
+
+        loop {
+            match parser
+                .parse(data, true)
```

Comment:
The `true` value here, belongs to the EOF flag, which represents wether more data will be received.
As we count with all bytes in memory for current use cases we assume bytes are present in memory.